krun/util.py

```diff
@@ -260,6 +260,22 @@ def check_and_parse_execution_results(stdout, stderr, rc, config,
             err_s += "Exception string: %s\n" % str(e)
         err_s += "return code: %d\n" % rc
         err_s += "stdout:\n%s\n%s\n%s\n\n" % (rule, stdout, rule)
+
+        # In instrumentation mode, stderr will have been written to disk for
+        # parsing. Read the last 50 lines back (it may be huge) so as to offer
+        # the user at-least something relating to the error.
+        if instrument:
+            from krun.vm_defs import INST_STDERR_FILE
```

Comment:
Should this `import` be at the top of the file?

Reply:
no, that's the circular import I was moaning about

Reply:
Ah, fine. Maybe add a comment to that effect? Even just `# Avoid circular import problems` or similar.

Reply:
Point taken.